File: android/src/main/kotlin/com/snowplowanalytics/snowplow_flutter_tracker/readers/configurations/NetworkConfigurationReader.kt
package com.snowplowanalytics.snowplow_flutter_tracker.readers.configurations

import com.snowplowanalytics.snowplow.configuration.NetworkConfiguration
import com.snowplowanalytics.snowplow.network.HttpMethod

class NetworkConfigurationReader(values: Map<String, Any>) {
    private val valuesDefault = values.withDefault { null }

    val endpoint: String by values
    val method: String? by valuesDefault

    fun toConfiguration(): NetworkConfiguration {
        if (method != null) {
            return NetworkConfiguration(
               endpoint,
               if ("get".equals(method, true)) { HttpMethod.GET } else { HttpMethod.POST }
            )
        } else {
            return NetworkConfiguration(endpoint)
        }
    }
}

File: ios/Classes/readers/configurations/NetworkConfigurationReader.swift
//
//  NetworkConfigurationReader.swift
//  FMDB
//
//  Created by Matus Tomlein on 03/01/2022.
//

import Foundation
import SnowplowTracker

struct NetworkConfigurationReader: Decodable {
    let endpoint: String
    let method: String?
}

extension NetworkConfigurationReader {
    func toConfiguration() -> NetworkConfiguration {
        if let m = method {
            return NetworkConfiguration(endpoint: endpoint, method: m == "get" ? .get : .post)
        } else {
            return NetworkConfiguration(endpoint: endpoint, method: .post)
        }
    }
}
